### ISSUE TITLE
[Core] Add learning rate decays

### ISSUE THREAD

**Issue by michaelschaarschmidt** [rlgraph/rlgraph#16]
Add an option to either/both:

i) Use an internal decay mechanism via global step + a learning_rate_spec as part of the update_spec
ii) Allow for an optional learning_rate parameter in update() to externally manipulate learning rates based on whatever scheme desired

Slight preference for ii) because it allows easier experimentation with irregular decay schemes.

**Comment by sven1977** [comment]
Completed as of version 0.5.0 (2019/05/23).

See FAQs (https://github.com/rlgraph/rlgraph/blob/master/FAQ.md#how-can-i-train-with-decayingchanging-parameters-like-learning-rates-over-time) for details on how to use this feature in any learning-rate, loss-function parameter, epsilon-exploration, etc.. via json configs or in your own Components.

Closing this issue.
